### PR TITLE
Merge 16.7 beta 5 to develop

### PR DESCRIPTION
**Pull request by jkmassel**
Merges all changes in the `16.7` beta series into `develop`

### CODE REVIEW

File: WordPress/Classes/Stores/StatsWidgetsStore.swift [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+        }
+        var widgetKind = ""
+        if widgetType == HomeWidgetTodayData.self, let stats = stats as? TodayWidgetStats {
+
+            widgetKind = WPHomeWidgetTodayKind
+
+            homeWidgetCache[siteID.intValue] = HomeWidgetTodayData(siteID: siteID.intValue,
+                                                                   siteName: blog.title ?? oldData.siteName,
+                                                                   url: blog.url ?? oldData.url,
+                                                                   timeZone: blogService.timeZone(for: blog),
```
Comment (Gio2018) [inline WordPress/Classes/Stores/StatsWidgetsStore.swift]: all the calls to `blogService.timeZone(for: blog)` should be changed to `blog.timeZone`

File: WordPress/Classes/Stores/StatsWidgetsStore.swift [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+                                                                   stats: stats) as? T
+
+
+        } else if widgetType == HomeWidgetAllTimeData.self, let stats = stats as? AllTimeWidgetStats {
+            widgetKind = WPHomeWidgetAllTimeKind
+
+            homeWidgetCache[siteID.intValue] = HomeWidgetAllTimeData(siteID: siteID.intValue,
+                                                                     siteName: blog.title ?? oldData.siteName,
+                                                                     url: blog.url ?? oldData.url,
+                                                                     timeZone: blogService.timeZone(for: blog),
```
Comment (Gio2018) [inline WordPress/Classes/Stores/StatsWidgetsStore.swift]: +1

File: WordPress/Classes/Stores/StatsWidgetsStore.swift [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+            }
+            let existingSite = currentData[blogID.intValue]
+
+            let siteURL = site.url ?? existingSite?.url ?? ""
+            let siteName = (site.title ?? siteURL).isEmpty ? siteURL : site.title ?? siteURL
+
+            var timeZone = existingSite?.timeZone ?? TimeZone.current
+
+            if let blog = blogService.blog(byBlogId: blogID) {
+                timeZone = blogService.timeZone(for: blog)
```
Comment (Gio2018) [inline WordPress/Classes/Stores/StatsWidgetsStore.swift]: +1

File: WordPress/Classes/Stores/StatsWidgetsStore.swift [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+    private func initializeHomeWidgetData<T: HomeWidgetData>(type: T.Type) -> [Int: T] {
+        let blogService = BlogService(managedObjectContext: ContextManager.shared.mainContext)
+
+        return blogService.visibleBlogsForWPComAccounts().reduce(into: [Int: T]()) { result, element in
+            if let blogID = element.dotComID,
+               let url = element.url,
+               let blog = blogService.blog(byBlogId: blogID) {
+                // set the title to the site title, if it's not nil and not empty; otherwise use the site url
+                let title = (element.title ?? url).isEmpty ? url : element.title ?? url
+                let timeZone = blogService.timeZone(for: blog)
```
Comment (Gio2018) [inline WordPress/Classes/Stores/StatsWidgetsStore.swift]: +1

**Review by Gio2018 — COMMENTED** [review]
Aside from the comments, the changes related to `StatsWidgetsStore` look ok to me. I encountered a build error on Gutenberg but I did not troubleshoot that part.

**Comment by jkmassel** [comment]
Thanks @Gio2018!!

**Review by JavonDavis — APPROVED** [review]
LGTM

**Pull request merged**